python/pyarrow/io.pxi

```diff
@@ -436,14 +440,22 @@ cdef class NativeFile(_Weakrefable):
     def read1(self, nbytes=None):
         """Read and return up to n bytes.
 
-        Alias for read, needed to match the BufferedIOBase interface.
+        A short result does not imply that EOF is imminent.
 
         Parameters
         ----------
         nbytes : int
             The maximum number of bytes to read.
         """
-        return self.read(nbytes=None)
+        if nbytes is None:
+            # The expectation when passing `nbytes=None` is not to read the
+            # entire file but to issue a single underlying read call up to
+            # a reasonable size (the use case being to read a bufferable
+            # amount of bytes, such as with io.TextIOWrapper).
+            nbytes = self._default_chunk_size
+        else:
+            nbytes = min(nbytes, self._default_chunk_size)
```

Comment:
Why are we limiting the read size to the chunk size when an explicit size is passed?

Reply:
Hmm, you're right that it may not be necessary, Python's IO stack is happily letting you read1() large sizes.